application/controllers/admin/SurveysGroupsController.php

```diff
@@ -194,16 +272,546 @@ public function surveySettings($id)
         $aData['jsData'] = [
             'sgid' => $id,
             'baseLinkUrl' => 'admin/surveysgroups/sa/surveysettings/id/'.$id,
-            'getUrl' => Yii::app()->createUrl('admin/globalsettings/sa/surveysettingmenues'),
+            'getUrl' => Yii::app()->createUrl(
+                'admin/surveysgroups/sa/surveysettingmenues',
+                array('id' => $id)
+            ),
             'i10n' => [
                 'Survey settings' => gT('Survey settings')
             ]
         ];
+        $aData['buttons'] = array(
+            'closebutton'=>array(
+                'url' => App()->createUrl('surveyAdministration/listsurveys', array('#' => 'surveygroups')),
+            ),
+        );
+        if (Permission::model()->hasSurveyGroupPermission($id, 'surveysettings', 'update')) {
+            $aData['buttons']['savebutton'] = array(
+                'form' => 'survey-settings-options-form'
+            );
+            $aData['buttons']['saveandclosebutton'] = array(
+                'form' => 'survey-settings-options-form'
+            );
+        }
         $aData['partial'] = $sPartial;
 
         $this->_renderWrappedTemplate('surveysgroups', 'surveySettings', $aData);
     }
 
+    /**
+     * Shown permissions list, allow to add user and group,
+     * No action done
+     * @param integer$id SurveysGroups id
+     */
+    public function permissions($id)
```

Comment:
`showPermissions`?

Reply:
Yep, can be

Reply:
Question : Create SurveyGroupsPermission controller or not ? If create `SurveyGroupsPermission` : i do it after (all url must be updated)